packages/gestalt/src/SheetMobile/ContentContainer.js

```diff
@@ -0,0 +1,91 @@
+// @flow strict
+import { type Node, useCallback, useState, useEffect, useRef } from 'react';
+import Box from '../Box.js';
+
+type Props = {|
+  children?: Node,
+  footer?: Node,
+  padding?: 'default' | 'none',
+  header: Node,
+|};
+
+export default function InternalPartialPageSheetMobile({
+  children,
+  header,
+  footer,
+  padding,
+}: Props): Node {
+  const [showTopShadow, setShowTopShadow] = useState(false);
+
+  const [showBottomShadow, setShowBottomShadow] = useState(false);
+
+  const contentRef = useRef<?HTMLElement>(null);
+
+  const updateShadows = useCallback(() => {
+    const target = contentRef.current;
+    if (!target) return;
+
+    const hasVerticalScrollbar = target?.clientHeight < target?.scrollHeight;
+    setShowTopShadow(hasVerticalScrollbar && target.scrollTop > 0);
+    setShowBottomShadow(
+      hasVerticalScrollbar && target.offsetHeight + target.scrollTop < target.scrollHeight,
+    );
+  }, []);
+
+  useEffect(() => {
+    updateShadows();
+  }, [updateShadows]);
+
+  useEffect(() => {
+    // When SheetMobile is full page displayed in mobile browser, the body scroll is still accessible. Here we disable to just allow the scrolling within Modal
+    if (!!window && window.body?.style?.overflow) {
+      window.body.style.overflow = 'hidden';
+    }
+    return () => {
+      if (!!window && window.body?.style?.overflow) {
+        window.body.style.overflow = 'auto';
```

Comment:
ditto here on assumptions about the previous value of this style

Reply:
This is a copy paste from Modal mobile... I figure out this was needed back when I developed it. Removing this adds a scrolling bug